src/chart/layout/mod.rs

```diff
@@ -0,0 +1,400 @@
+            pub fn [<clear_ $name _text>](
+                &mut self,
+            ) -> Result<&mut Self, DrawingAreaErrorKind<DB::ErrorType>> {
+                self.[<set_ $name _text>]("")?;
+                Ok(self)
+            }
+            #[doc = "Set the style of the `" $name "` label's text."]
+            pub fn [<set_ $name _style>]<Style: IntoTextStyle<'b>>(
+                &mut self,
+                style: Style,
+            ) -> Result<&mut Self, DrawingAreaErrorKind<DB::ErrorType>> {
+                Self::set_style(self.root_area, &mut self.$name, style);
+                self.[<recompute_ $name _size>]()?;
+                Ok(self)
+            }
+            #[doc = "Set the margin of the `" $name "` container. If `margin` is a single"]
+            #[doc = "number, that number is used for all margins. If `margin` is a tuple `(vert,horiz)`,"]
+            #[doc = "then the top and bottom margins will be `vert` and the left and right margins will"]
+            #[doc = "be `horiz`. To set each margin separately, use a [`Margin`] struct or a four-tuple."]
+            pub fn [<set_ $name _margin>]<M: Into<Margin<f32>>>(
+                &mut self,
+                margin: M,
+            ) -> Result<&mut Self, DrawingAreaErrorKind<DB::ErrorType>> {
+                self.nodes.[<set_ $name _margin>](margin)?;
+                Ok(self)
+            }
+            #[doc = "Gets the margin of the `" $name "` container."]
+            pub fn [<get_ $name _margin>](
+                self,
+            ) -> Result<Margin<f32>, DrawingAreaErrorKind<DB::ErrorType>> {
+                Ok(self.nodes.[<get_ $name _margin>]()?)
+            }
+
+
+        }
+    };
+}
+
+/// Hold the text and the style of a chart label
+struct Label<'a> {
+    text: Option<String>,
+    style: TextStyle<'a>,
+}
+
+/// The helper object to create a chart context, which is used for the high-level figure drawing.
+/// With the help of this object, we can convert a basic drawing area into a chart context, which
+/// allows the high-level charting API being used on the drawing area.
+pub struct ChartLayout<'a, 'b, DB: DrawingBackend> {
+    root_area: &'a DrawingArea<DB, Shift>,
+    chart_title: Label<'b>,
+    top_label: Label<'b>,
+    bottom_label: Label<'b>,
+    left_label: Label<'b>,
+    right_label: Label<'b>,
+    nodes: ChartLayoutNodes,
+}
+
+impl<'a, 'b, DB: DrawingBackend> ChartLayout<'a, 'b, DB> {
+    /// Create a chart builder on the given drawing area
+    /// - `root`: The root drawing area
+    /// - Returns: The chart layout object
+    pub fn new(root: &'a DrawingArea<DB, Shift>) -> Self {
+        Self {
+            root_area: root,
+            chart_title: Label {
+                text: None,
+                style: TextStyle::from(("serif", 40.0).into_font()),
+            },
+            top_label: Label {
+                text: None,
+                style: TextStyle::from(("serif", 25.0).into_font()),
```

Comment:
Actually, you can just use `("serif", 25.0).into()`.

Reply:
Also, if we want to optimize the default appearance, I think we should use relative sizing.